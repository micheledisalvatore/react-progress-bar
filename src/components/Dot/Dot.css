.dot {
    background-color: transparent;
    border: none;
    padding: 0;
    position: relative;
    z-index: 1;
}

.dot__image {
    border: 0.6em solid #DBDBDB;
    border-radius: 50%;
    height: 2.5em;
    width: 2.5em;
}

.dot__image--is-active {
    border-color: #5131D7;
    transition: border 0s linear;
    transition-delay: 500ms;
}