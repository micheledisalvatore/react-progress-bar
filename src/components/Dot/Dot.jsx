import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './Dot.css';

export const Dot = ({isActive, isClickable, onClick, step}) => {
  const dotClassName = classNames('dot__image', {
    'dot__image--is-active': isActive,
  });

  const handleClick = e => {
    e.preventDefault();

    onClick(step);
  };

  return <button className="dot" onClick={handleClick} disabled={!isClickable}>
    <svg className={dotClassName} viewBox="0 0 32 32"/>
  </button>
};

Dot.propTypes = {
  isActive: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  step: PropTypes.oneOf([1,2,3,4,5]).isRequired,
};

Dot.defaultProps = {
  isActive: false,
};
