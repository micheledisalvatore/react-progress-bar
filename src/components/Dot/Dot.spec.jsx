import React from 'react';
import {shallow} from 'enzyme';

import {Dot} from './Dot';

describe('Given a Dot component', () => {
  describe('when it is rendered', () => {
    const STEP = 1;
    const onClickMock = jest.fn();
    const requiredProps = {
      step: STEP,
      onClick: onClickMock,
    };
    const component = <Dot {...requiredProps} />;
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(component);
    });

    it('should match the snapshot', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should render with NOT the active class', () => {
      expect(wrapper.find('svg')).not.toHaveClassName('dot__image--is-active');
    });

    describe('and the user clicks on it', () => {
      beforeEach(() => {
        wrapper.simulate('click', {preventDefault: jest.fn()});
      });

      it('should call onClick with the current step', () => {
        expect(onClickMock).toHaveBeenCalledWith(STEP);
      });
    });

    describe('and it receives isActive prop', () => {
      beforeEach(() => {
        wrapper.setProps({
          isActive: true,
        });
      });

      it('should render with the active class', () => {
        expect(wrapper.find('svg')).toHaveClassName('dot__image--is-active');
      });
    });
  });
});
