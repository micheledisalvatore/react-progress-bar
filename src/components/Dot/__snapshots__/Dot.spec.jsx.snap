// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Dot component when it is rendered should match the snapshot 1`] = `
<button
  className="dot"
  onClick={[Function]}
>
  <svg
    className="dot__image"
    viewBox="0 0 32 32"
  />
</button>
`;
